src/api/api.ts

```diff
@@ -1,69 +1,175 @@
+  (window as any).universalRequest = universalRequest;
+}
+
+async function makeWebRequest<ResponsePayload = unknown, RequestPayload = unknown>(
+  method: HttpMethod,
+  path: string,
+  payload: RequestPayload | undefined,
+  responseParser: ResponseParser,
+) {
+  console.log(`WEB FETCH REQUEST: ${method} ${path}`);
+  const headers: HeadersInit =
+    payload instanceof FormData
+      ? {}
+      : {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        };
+  const response = await fetch(path, {
+    method,
+    headers,
+    body: payload instanceof FormData ? payload : payload ? JSON.stringify(payload) : undefined,
+  });
+  console.log('WEB FETCH RESPONSE STATUS:', response.status);
+  if (!response.ok) {
+    throw new Error(`Failed to ${method} ${path}: ${response.statusText}`);
+  }
+  switch (responseParser) {
+    case 'JSON':
+      return (await response.json()) as ResponsePayload;
+    case 'ArrayBuffer':
+      return (await response.arrayBuffer()) as ResponsePayload;
   }
 }
+
+async function makeTauriRequest<ResponsePayload = unknown, RequestPayload = unknown>(
+  path: string,
+  method: HttpMethod,
+  payload: RequestPayload | undefined,
+  responseParser: ResponseParser,
+) {
+  const url = REFSTUDIO_HOST + path;
+  console.log(`TAURI FETCH REQUEST: ${method} ${url}`);
+  const responseType = responseParser === 'JSON' ? ResponseType.JSON : ResponseType.Binary;
+  const response = await tauriFetch(url, {
+    method,
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-explicit-any
+    body: payload ? Body.json(payload as any) : undefined,
+    responseType,
+  });
+  console.log('TAURI FETCH RESPONSE STATUS', response.status);
+  if (!response.ok) {
+    throw new Error(`Failed to ${method} ${path}: ${response.status} - ${JSON.stringify(response.data)}`);
+  }
+  switch (responseParser) {
+    case 'JSON':
+      console.log('TAURI FETCH RESPONSE DATA', response.data);
+      return response.data as ResponsePayload;
+    case 'ArrayBuffer':
+      console.log('TAURI FETCH RESPONSE DATA', response.data);
+      return response.data as ResponsePayload;
+  }
+}
+
+/** Issue a FILE Upload using either web fetch or Tauri fetch */
+export async function universalPutFile(path: string, filePath: string, content: string | ArrayBuffer) {
+  if (import.meta.env.VITE_IS_WEB) {
+    const formData = new FormData();
+    formData.append('file', new File([content], filePath));
+    return fetch(path, { method: 'PUT', body: formData });
+  }
+
+  // TAURI
+  const url = REFSTUDIO_HOST + path;
+  const buffer = typeof content === 'string' ? new TextEncoder().encode(content) : new Uint8Array(content);
+  const body = Body.form({
+    file: {
+      // either a path or an array buffer of the file contents.
+      // We can't use a path bc Tauri check if file exists and has read permissions. Also, will send the file content.
```

Comment:
Is there a bug you can reference here? This is surprising behavior.

Reply:
I don't have it. I've got 422 Unprocessable Entity from the python code when this field was missing.
I will reproduce tomorrow and send you.